### ISSUE TITLE
Dry Test / Live - trades are sold after buy =1(entry) > sell = 1 > buy =1(exit) 

### ISSUE THREAD

**Issue by creslinux** [freqtrade/freqtrade#1042]
I've moved my QA into dry test / live. 
There are a number of issues seen that directly impact profitability which are also confirmed anecdotally by users in Slack.

Test scenarion 53 pairs, max trades 53, strategy returns 0 on even minutes, 1  on odd - no delay.

Ill paraphrase but the workflow of the bot in dry test quickly becomes thus. 

**odd minute - sell =1**
1) tests though loop of pairs while sell = 1  / buy = 0, maybe 1-2 seconds per pair.

**even minute - buy = 1**
2) 
 - opens first trade. Time to process 1 second
     tends to trade 1 
 - opens second trade : Time to process 4 seconds
    tends to trade 1
    tends to trade 2
 - opens third trade : Time to process 6 seconds
   tends to trade 1
   tends to trade 2
   tends to trade 3
......
trade 7 takes 12 seconds to open. No more pairs are open.
<img width="758" alt="screen shot 2018-07-18 at 5 02 22 pm" src="https://user-images.githubusercontent.com/34645187/42896731-607d5f14-8aad-11e8-8870-ad9b528c09c7.png">

3) The bot spends an ever increasing time managing open trades and during this time does not process the "Other pairs" 

**Even minute - Sell = 1**
4) The bot notices the change,  
        - as now no longer solely focuses on the open pairs
        - processes other pairs from the 53
        - none of the 7 pairs open i trade are sold...... **They should be sold**

**even minute - buy = 1**
5) The 7 open trades now close ,, 

first trade closes on exacly even minute when buy =1
<img width="577" alt="screen shot 2018-07-18 at 5 10 49 pm" src="https://user-images.githubusercontent.com/34645187/42896802-8ca2fb76-8aad-11e8-9e65-962ad50b7fa1.png">

others close very soon after (2 seconds) (but not when they are sell-1, when flipped back to buy =1
<img width="649" alt="screen shot 2018-07-18 at 5 12 02 pm" src="https://user-images.githubusercontent.com/34645187/42896857-ba2ec160-8aad-11e8-9859-b480bceb3f73.png">

6) New trades begin to open again - as per step 2. 
     It is the same pairs to open. 
<img width="669" alt="screen shot 2018-07-18 at 5 13 11 pm" src="https://user-images.githubusercontent.com/34645187/42896922-eb93d1dc-8aad-11e8-9d09-5c728face143.png">

7) repeat. 

In terms of issues there are a few

The most important for this issue is trades are not sold on sell=1 but when flipping from buy=1 to  sell =1 and back to buy =1. 
This ensures poor trading. 

A video of the issue. 

https://highfrequencybot.slack.com/files/UAU4J8LU9/FBTQFU431/ft_drytest.mp4

**Comment by creslinux** [comment]
Observed behaviour  

<img width="640" alt="screen shot 2018-07-18 at 7 14 39 pm" src="https://user-images.githubusercontent.com/34645187/42902715-1bf297e4-8abf-11e8-918f-ac056ba3bc4b.png">

**Comment by creslinux** [comment]
observed behaviour 
<img width="650" alt="screen shot 2018-07-18 at 7 17 32 pm" src="https://user-images.githubusercontent.com/34645187/42902847-717414a4-8abf-11e8-8b30-1af9ca106289.png">

**Comment by creslinux** [comment]
on closer inspection trades are sold on time, the update the trades db is delayed 
 
a "Sell" entry in exchange.sell _dry_run_open_orders is made on time, as the odd minute arrives. 
processed a the front of the queue. 

```
buy sell is  False True
closed with  {'pair': 'ZEC/BTC', 'price': 0.027536, 'amount': 1.807141824490386, 'type': 'limit', 'side': 'sell', 'remaining': 0.0, 'datetime': '2018-07-18T20:41:00.099324+00:00', 'status': 'closed'}
2018-07-18 20:41:00,792 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell, 'exchange': 'Binance', 'pair': 'ZEC/BTC', 'gain': 'loss', 'market_url': 'https://www.binance.com/tradeDetail.html?symbol=ZEC_BTC', 'limit': 0.027536, 'amount': 1.807141824490386, 'open_rate': 0.027668, 'current_rate': 0.027536, 'profit_amount': -0.0003383, 'profit_percent': -0.00675932, 'profit_fiat': -2.469031805, 'stake_currency': 'BTC', 'fiat_currency': 'USD'}
----------------we are in strategy 20:41:02.583836
sell = 1
```
